DataMigration/PostgresDB/DataProvider.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System.Text;
+using System.Collections.Generic;
+using DataMigration.PostgresDB.Entities;
+using Npgsql;
+
+namespace DataMigration.PostgresDB
+{
+    using Logger;
+    using Logger.Enums;
+
+    partial class DataProvider : IDataProvider, ILogSupporting
+    {
+        private readonly string _connectionString;
+
+        public DataProvider(string connectionString)
+        {
+            LogEventHappened?.Invoke($"Creating new DataProvider. Connection string: {connectionString}", LogLevel.Debug);
+            _connectionString = connectionString;
+            LogEventHappened?.Invoke($"Opening connection...", LogLevel.Debug);
+            LogEventHappened?.Invoke($"Connection opened!", LogLevel.Debug);
+        }
+
+        public event MakeLog LogEventHappened;
+
+        public  NpgsqlConnection GetOpenedConnection()
+        {
+            LogEventHappened("Creating and opening new connection to Postgres Db...", LogLevel.Debug);
+            var connection = new NpgsqlConnection(_connectionString);
+            connection.Open();
+            LogEventHappened("Success! Connection opened!", LogLevel.Debug);
+            return connection;
+        }
+
+        public List<HistoricalOcrData> GetHistoricalOcrData(int? count = null, bool deleteAfterSelect = false)
+        {
+            LogEventHappened?.Invoke($"Started getting historical ocr data. Count:  {(count == null ? "all" : count.ToString())}", LogLevel.Debug);
+
+            string commandText = "SELECT * FROM historical_ocr_data";
+            if (count != null)
+            {
+                commandText += string.Format(" LIMIT {0}", count);
+            }
+
+            List<HistoricalOcrData> result = new List<HistoricalOcrData>();
+
+            using (var connection = GetOpenedConnection())
+            using (var command = new NpgsqlCommand(commandText, connection))
+            {
+                LogEventHappened?.Invoke($"Executing query...", LogLevel.Debug);
+                NpgsqlDataReader reader = command.ExecuteReader();
+                LogEventHappened?.Invoke($"Done!", LogLevel.Debug);
+                while (reader.Read())
+                {
+                    int tenantId = reader.GetInt32(0);
+                    string fullFilePath = reader.GetString(1);
+                    int statusId = reader.GetInt32(2);
+                    string errorMessage = reader.GetValue(3)?.ToString();
+                    string data = reader.GetString(4);
+                    object createdAt = reader.GetValue(5);
+                    object updatedAt = reader.GetValue(6);
+                    result.Add(new HistoricalOcrData(tenantId, fullFilePath, statusId, errorMessage, data, createdAt, updatedAt));
+                }
+            }
+
+
+            LogEventHappened?.Invoke($"Data is successfully selected! Actual total count: {result.Count}", LogLevel.Debug);
+            if (deleteAfterSelect)
```

Comment:
bad solution. if some exception occurs before pushing messages to rabbit (e.g. connection to rabbit can't be established) data will be lost